### ISSUE TITLE
Inconsistent behaviour of printf %#0(insert number)x with different types

### ISSUE THREAD

**Issue by Sebi2020** [golang/go#28627]
#### What did you do?
I wanted print a padded (zeros)  and prepended (0x) string character as a hex number.

[Link to example on play.golang.org](https://play.golang.org/p/1ZTAooO_0Y2)

#### What did you expect to see?
The same result for a string character and an integer.
The string printed as an hexadecimal number should be printed with a prepended '0x'.

#### What did you see instead?
```
> A hex Number (a int) with padding:    0x0000000030
> A hex Number (a string) with padding: 0000000x30
> 
> Program exited.
```

#### System details

```
go version go1.11.2 linux/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/sebi/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/sebi/go/"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
GOROOT/bin/go version: go version go1.11.2 linux/amd64
GOROOT/bin/go tool compile -V: compile version go1.11.2
uname -sr: Linux 4.15.0-38-generic
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
/lib/x86_64-linux-gnu/libc.so.6: GNU C Library (Ubuntu GLIBC 2.23-0ubuntu10) stable release version 2.23, by Roland McGrath et al.
gdb --version: GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
```

**Comment by martisch** [comment]
This seems to be working as documented:
* int: # for 0x for hex (%#x) and 0 for pad with leading zeros rather than spaces
* string: %x base 16, lower-case, two characters *per byte* then # for 0x for hex (%#x)  for each byte and pad the string with leading zeros rather than spaces.

The difference is that the verb %x applies to each byte in strings but the rest of the modifiers are not.

**Comment by Sebi2020** [comment]
But it does not add an '0x' for each byte:
(Example)[https://play.golang.org/p/kTuXoeOZ5mo]
```
> 00000x3031
```
As far as I understood you it should print something like:

```
> 0x00000x300x31
```

**Comment by martisch** [comment]
you are right it i was confusing this with ' ' where it applies per byte and is mentioned explicitly "% x" in documentation.

Referring back to the original issue of 0 padding I think its still in line with documentation that padding of 0 applies to the whole string after it has been formatted?

**Comment by Sebi2020** [comment]
> Referring back to the original issue of 0 padding I think its still in line with documentation that padding of 0 > applies to the whole string after it has been formatted? Could maybe made explicit.

I cannot find this in the documentation for printf.

This is what I've found:
> 
> ```
>  ' '	(space) leave a space for elided sign in numbers (% d);
> 	put spaces between bytes printing strings or slices in hex (% x, % X)
> 0	pad with leading zeros rather than spaces;
> 	for numbers, this moves the padding after the sign
> 
> ```
It does only say what happens if you pad it with spaces, so I agree it should be more explicitly stated what happens if you use zeros instead of spaces.

**Comment by bcmills** [comment]
CC @robpike

**Comment by robpike** [comment]
Padding applies to the whole result.
